src/shared/components/case-editor/services/wizard-page-field-to-case-field.mapper.ts

```diff
@@ -0,0 +1,88 @@
+import { WizardPageField } from '../domain';
+import { CaseField } from '../../../domain';
+import { ComplexFieldOverride } from '../domain/wizard-page-field-complex-override.model';
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class WizardPageFieldToCaseFieldMapper {
+
+  mapAll(wizardPageFields: WizardPageField[], caseFields: CaseField[]): CaseField[] {
+    return wizardPageFields.map(wizardField => {
+      return this.map(wizardField, caseFields);
+    });
+  }
+
+  private map(wizardPageField: WizardPageField, caseFields: CaseField[]): CaseField {
+
+    let caseField: CaseField = caseFields.find(e => e.id === wizardPageField.case_field_id);
+    caseField.id = wizardPageField.case_field_id;
+    caseField.wizardProps = wizardPageField;
+    caseField.display_context = wizardPageField.display_context;
+    caseField.order = wizardPageField.order;
+
+    if (wizardPageField.complex_field_overrides && wizardPageField.complex_field_overrides.length > 0) {
+      wizardPageField.complex_field_overrides.forEach((override: ComplexFieldOverride) => {
+        this.processComplexFieldOverride(override, caseField, caseFields);
+      });
+    }
+
+    return caseField;
+  }
+
+  private processComplexFieldOverride(override: ComplexFieldOverride, case_field: CaseField, caseFields: CaseField[]) {
+    const caseFieldIds = override.complex_field_element_id.split('.');
+    let case_field_leaf: CaseField;
+
+    if (this.isCollectionOfComplex(case_field)) {
+      const [_, ...tail] = caseFieldIds;
+      case_field_leaf = this.getCaseFieldLeaf(tail, case_field.field_type.collection_field_type.complex_fields);
+    } else {
+      case_field_leaf = this.getCaseFieldLeaf(caseFieldIds, caseFields);
+    }
+
+    if (override.display_context !== 'HIDDEN') {
```

Comment:
As of now, the backend sends a ComplexFieldOverride with display context 'HIDDEN' for all the subfields which don't have an override. Isn't this a bit redundant?
Can't the backend only send ComplexFieldOverride for the subfields that have an override. And the UI could assume that all the other complex subfields are hidden?

Reply:
We have decided with Valentin we don't want to add more logic to the front-end if not necessary. Hence the logic is in the BE.